packages/grid_client/tests/modules/applications/casperlabs.test.ts

```diff
@@ -0,0 +1,242 @@
+  expect(result[0].description).toBe(description);
+  expect(result[0].rootfs_size).toBe(bytesToGB(rootfsSize));
+  expect(result[0].mounts[0]["name"]).toBe(diskName);
+  expect(result[0].mounts[0]["size"]).toBe(bytesToGB(diskSize));
+  expect(result[0].mounts[0]["mountPoint"]).toBe(mountPoint);
+  expect(result[0].mounts[0]["state"]).toBe("ok");
+
+  const backends = ["http://[" + result[0].planetary + "]:80"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw: GatewayNameModel = {
+    name: name,
+    node_id: GatewayNode.nodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  const gatewayRes = await gridClient.gateway.deploy_name(gw);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(GatewayNode.nodeId);
+
+  const gatewayResult = await gridClient.gateway.getObj(gw.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name);
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  const site = "https://" + gatewayResult[0].domain;
+  let reachable = false;
+
+  for (let i = 0; i < 180; i++) {
+    const wait = await setTimeout(5000, "Waiting for gateway to be ready");
+    log(wait);
+
+    await axios
+      .get(site)
+      .then(res => {
+        log("gateway is reachable");
+        log(res.status);
+        log(res.statusText);
+        log(res.data);
+        expect(res.status).toBe(200);
+        expect(res.statusText).toBe("OK");
+        expect(res.data).toContain("Your Casper node is now running succesfully on the ThreeFold Grid.");
+        reachable = true;
+      })
+      .catch(() => {
+        log("gateway is not reachable");
+      });
+    if (reachable) {
+      break;
+    } else if (i == 180) {
+      throw new Error("Gateway is unreachable after multiple retries");
+    }
+  }
+});
+
+afterEach(async () => {
+  const vmNames = await gridClient.machines.list();
+  for (const name of vmNames) {
+    const res = await gridClient.machines.delete({ name });
+    log(res);
+    expect(res.created).toHaveLength(0);
+    expect(res.updated).toHaveLength(0);
+    expect(res.deleted).toBeDefined();
+  }
+
+  const gwNames = await gridClient.gateway.list();
+  for (const name of gwNames) {
+    const res = await gridClient.gateway.delete_name({ name });
+    log(res);
+    expect(res.created).toHaveLength(0);
+    expect(res.updated).toHaveLength(0);
+    expect(res.deleted).toBeDefined();
+  }
+});
+
+afterAll(async () => {
+  return await gridClient.disconnect();
+}, 130000);
```

Comment:
No need to use afterAll as it's only one test. Can we combine both in afterEach?

Reply:
Done in [d87b135](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/2784/commits/d87b135083ec70715e0816363317dc54c7e9a16e), and a timeout is necessary because deleting and disconnecting take some time.